2018/python/rules_engine.py

```diff
@@ -635,7 +635,10 @@ def v628_1(self):
 		unless an ethnicity is provided in Ethnicity of Applicant or Borrower: Free Form Text Field for Other Hispanic or Latino."""
 		field = "app_eth_1"
 		edit_name = "v628_1"
-		fail_df = self.lar_df[~(self.lar_df.app_eth_1.isin(("1","11", "12", "13", "14", "2", "3","4")))|((self.lar_df.app_eth_free=="")&(self.lar_df.app_eth_1==""))]
+		fail_df = self.lar_df[((~self.lar_df.app_eth_1.isin(("1","11", "12", "13", "14", "2", "3","4"))) & 
+								(self.lar_df.app_eth_free == ''))|
+							  ((self.lar_df.app_eth_1.isin(("1","11", "12", "13", "14", "2", "3","4"))) & 
+								(self.lar_df.app_eth_free != ''))]
```

Comment:
I think code 14 needs to be removed from this list to make the logic check work for a populated free text field.

Reply:
Sure, we can do this - I know that code 14 is included in the FIG as a valid value for applicant ethnicity, and so I was hesitant to remove it. I can do so if you think it would be best this way. 

The code is working now as is.